drivers/mipi_dbi/mipi_dbi_smartbond.c

```diff
@@ -0,0 +1,610 @@
+/*
+ * Copyright (c) 2023 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT  renesas_smartbond_mipi_dbi
+
+#include <zephyr/drivers/mipi_dbi.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/device_runtime.h>
+#include <DA1469xAB.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/smartbond_clock_control.h>
+#include <zephyr/drivers/display.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/drivers/spi.h>
+#include <da1469x_lcdc.h>
+#include <da1469x_pd.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(smartbond_mipi_dbi, CONFIG_MIPI_DBI_LOG_LEVEL);
+
+#define SMARTBOND_IRQN		DT_INST_IRQN(0)
+#define SMARTBOND_IRQ_PRIO  DT_INST_IRQ(0, priority)
+
+#define PINCTRL_STATE_READ  PINCTRL_STATE_PRIV_START
+
+#define MIPI_DBI_SMARTBOND_IS_READ_SUPPORTED \
+				DT_INST_NODE_HAS_PROP(0, spi_dev)
+
+#define LCDC_SMARTBOND_CLK_DIV(_freq)                           \
+    ((32000000U % (_freq)) ? (96000000U / (_freq)) : (32000000U / (_freq)))
+
+#define MIPI_DBI_SMARTBOND_IS_PLL_REQUIRED \
+			!!(32000000U % DT_PROP(DT_CHOSEN(zephyr_display), mipi_max_frequency))
+
+#define MIPI_DBI_SMARTBOND_IS_TE_ENABLED \
+			DT_INST_PROP_OR(0, te_enable, 0)
+
+#define MIPI_DBI_SMARTBOND_IS_DMA_PREFETCH_ENABLED \
+			DT_INST_PROP_OR(0, dma_prefetch, 0)
+
+#define MIPI_DBI_SMARTBOND_IS_RESET_REQUIRED \
+			DT_INST_NODE_HAS_PROP(0, reset_gpios)
+
+#define LCDC_LAYER0_OFFSETX_REG_SET_FIELD(_field, _var, _val) \
+	((_var)) = \
+	((_var) & ~(LCDC_LCDC_LAYER0_OFFSETX_REG_ ## _field ## _Msk)) |	\
+	(((_var) << LCDC_LCDC_LAYER0_OFFSETX_REG_ ## _field ## _Pos) & \
+	LCDC_LCDC_LAYER0_OFFSETX_REG_ ## _field ## _Msk)
+
+struct mipi_dbi_smartbond_data {
+	/* Provide mutual exclusion when a display operation is requested. */
+	struct k_sem device_sem;
+	/* Provide synchronization between task return and ISR firing */
+	struct k_sem sync_sem;
+	/* Flag indicating whether or not an underflow took place */
+	bool underflow_flag;
+	/* Layer settings */
+	lcdc_smartbond_layer_cfg layer;
+};
+
+struct mipi_dbi_smartbond_config {
+	/* Reference to device instance's pinctrl configurations */
+	const struct pinctrl_dev_config *pcfg;
+	/* Reset GPIO */
+	const struct gpio_dt_spec reset;
+	/* Host controller's timing settings */
+	lcdc_smartbond_timing_cfg timing_cfg;
+	/* Background default color configuration */
+	lcdc_smartbond_bgcolor_cfg bgcolor_cfg;
+};
+
+/* Mark the device is is progress and so it's not allowed to enter the sleep state. */
+static void mipi_dbi_pm_get(const struct device *dev)
```

Comment:
Not sure if this will work for your use case, but `pm_policy_state_lock_get` might also be helpful here

Reply:
Given our `pm_state_set` SoC implementation, this should not work here. Thanks for pointing this out, either way.